File: ios/Noke/Noke.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Noke, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(openNoke)
@end

File: ios/Noke/Noke.swift
import UIKit
@objc(Noke)
class Noke: NSObject {
    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b)
    }
    
    @objc func openNoke() {
        DispatchQueue.main.async {
            let bundle: Bundle? = Bundle.main.path(forResource: "Resources", ofType: "bundle")
                .flatMap(Bundle.init(path:))
            print("Path `\(bundle)")
//            let rootVC = UIApplication.shared.keyWindow?.rootViewController
//            let vc = NokeViewController(nibName: "NokeView", bundle:bundle)
//            rootVC?.present(vc, animated: true, completion: nil)
        }
    }
    
}
